enterprise/pulsar.py

```diff
@@ -19,29 +19,28 @@
 except:
     import pickle
 
+logger = logging.getLogger(__name__)
+
 try:
     import libstempo as t2
 except ImportError:
-    print("Ooh, no libstempo?")
+    logger.warning("libstempo not installed. Will use PINT instead.")  # pragma: no cover
```

Comment:
@nanograv/enterprise These are not really needed if we have both `libstempo` and `PINT` as dependencies. Do we want to keep them both or have one as optional?

Reply:
I'm glad you changed them to `logger` warnings. I think we could make the case either way. People only need one of the packages to run enterprise. I think the only dependency on either package right now is within `enterprise.Pulsar`, so users can get a way with only using one. I guess that is a pretty rare situation for software. Maybe we just raise an `ImportError` if neither are present?

Reply:
Well I think there are 3 options:
1. Don't require either as base dependencies and have a `pip install enterprise-pulsar[libstempo]` and `pip install enterprise-pulsar[pint]` path. I don't really like this because it could lead to broken installs because the package won't really work without one or the other.
2. Choose one as the default and add the other as an extra. Kind of like this one but not sure which should be default
3. Leave both as it is now. This is fine too

Reply:
Actually the whole `Pulsar` function/class is a pretty bad design. I don't think I knew much about class polymorphism at the time. Its too much to refactor it for this PR but it could definitely be better.